zk/stages/stage_sequence_execute_transactions.go

```diff
@@ -158,7 +160,13 @@ func attemptAddTransaction(
 		return nil, nil, txCounters, overflowCounters, nil
 	}
 
-	gasPool := new(core.GasPool).AddGas(transactionGasLimit)
+	// if not normalcy we want to create a gas pool per transaction (zkevm block gas limit is infinite), if normalcy create a pool per block.
+	var gasPool *core.GasPool
+	if !cfg.chainConfig.IsNormalcy(blockContext.BlockNumber) {
+		gasPool = new(core.GasPool).AddGas(transactionGasLimit)
```

Comment:
This will allow the pool to go higher than it should.  
- Starting pool = 30m gas
- TX1 uses 15m gas which drops the pool to 15m
- Next tx we add 30m to the pool again so it's now 45m which is too much

When not in normalcy we need to completely refresh the pool back to 30m each time

Reply:
Ignore me, I misread this :)